src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -124,7 +124,7 @@ ucs_status_t uct_ib_mlx5_get_cq(struct ibv_cq *cq, uct_ib_mlx5_cq_t *mlx5_cq)
         return UCS_ERR_UNSUPPORTED;
     }
 
-    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
+    mlx5_cq->cqe_size_log = __ucs_ilog2_u32(cqe_size);
```

Comment:
what was the warning?

Reply:
```
Error: CLANG_WARNING:
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:128:25: warning: The result of the left shift is undefined due to shifting by '32', which is greater or equal to the width of type 'int'
#    ucs_assert_always((1<<mlx5_cq->cqe_size_log) == cqe_size);
#                        ^ ~~~~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/debug/assert.h:26:25: note: expanded from macro 'ucs_assert_always'
#        if (!ucs_likely(_expression)) { \
#                        ^~~~~~~~~~~
ucx-1.6.0/src/ucs/sys/compiler_def.h:151:53: note: expanded from macro 'ucs_likely'
##define ucs_likely(x)              __builtin_expect(x, 1)
#                                                    ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:98:9: note: Assuming 'status' is equal to UCS_OK
#    if (status != UCS_OK) {
#        ^~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:98:5: note: Taking false branch
#    if (status != UCS_OK) {
#    ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:122:5: note: Taking false branch
#    if (ret != 0) {
#    ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is true
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^
ucx-1.6.0/src/ucs/arch/bitops.h:23:5: note: expanded from macro 'ucs_ilog2'
#    __builtin_constant_p(_n) ? (        \
#    ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming 'cqe_size' is >= 1
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:24:14: note: expanded from macro 'ucs_ilog2'
#             (_n) < 1 ? 0 :             \
#             ^~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:24:14: note: expanded from macro 'ucs_ilog2'
#             (_n) < 1 ? 0 :             \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:25:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 63) ? 63 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:25:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 63) ? 63 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:26:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 62) ? 62 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:26:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 62) ? 62 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:27:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 61) ? 61 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:27:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 61) ? 61 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:28:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 60) ? 60 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:28:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 60) ? 60 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:29:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 59) ? 59 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:29:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 59) ? 59 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:30:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 58) ? 58 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:30:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 58) ? 58 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:31:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 57) ? 57 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:31:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 57) ? 57 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:32:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 56) ? 56 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:32:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 56) ? 56 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:33:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 55) ? 55 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:33:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 55) ? 55 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:34:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 54) ? 54 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:34:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 54) ? 54 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:35:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 53) ? 53 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:35:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 53) ? 53 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:36:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 52) ? 52 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:36:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 52) ? 52 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:37:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 51) ? 51 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:37:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 51) ? 51 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:38:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 50) ? 50 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:38:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 50) ? 50 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:39:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 49) ? 49 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:39:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 49) ? 49 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:40:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 48) ? 48 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:40:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 48) ? 48 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:41:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 47) ? 47 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:41:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 47) ? 47 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:42:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 46) ? 46 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:42:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 46) ? 46 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:43:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 45) ? 45 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:43:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 45) ? 45 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:44:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 44) ? 44 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:44:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 44) ? 44 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:45:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 43) ? 43 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:45:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 43) ? 43 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:46:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 42) ? 42 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:46:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 42) ? 42 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:47:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 41) ? 41 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:47:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 41) ? 41 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:48:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 40) ? 40 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:48:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 40) ? 40 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:49:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 39) ? 39 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:49:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 39) ? 39 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:50:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 38) ? 38 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:50:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 38) ? 38 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:51:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 37) ? 37 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:51:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 37) ? 37 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:52:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 36) ? 36 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:52:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 36) ? 36 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:53:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 35) ? 35 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:53:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 35) ? 35 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:54:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 34) ? 34 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:54:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 34) ? 34 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is false
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:55:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 33) ? 33 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is false
ucx-1.6.0/src/ucs/arch/bitops.h:55:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 33) ? 33 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: Assuming the condition is true
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#                            ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/arch/bitops.h:56:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 32) ? 32 : \
#             ^~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:29: note: '?' condition is true
ucx-1.6.0/src/ucs/arch/bitops.h:56:14: note: expanded from macro 'ucs_ilog2'
#             (_n) & (1ULL << 32) ? 32 : \
#             ^
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:127:5: note: The value 32 is assigned to field 'cqe_size_log'
#    mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/uct/ib/mlx5/ib_mlx5.c:128:25: note: The result of the left shift is undefined due to shifting by '32', which is greater or equal to the width of type 'int'
#    ucs_assert_always((1<<mlx5_cq->cqe_size_log) == cqe_size);
#                        ^ ~~~~~~~~~~~~~~~~~~~~~
ucx-1.6.0/src/ucs/debug/assert.h:26:25: note: expanded from macro 'ucs_assert_always'
#        if (!ucs_likely(_expression)) { \
#                        ^~~~~~~~~~~
ucx-1.6.0/src/ucs/sys/compiler_def.h:151:53: note: expanded from macro 'ucs_likely'
##define ucs_likely(x)              __builtin_expect(x, 1)
#                                                    ^
#  126|   
#  127|       mlx5_cq->cqe_size_log = ucs_ilog2(cqe_size);
#  128|->     ucs_assert_always((1<<mlx5_cq->cqe_size_log) == cqe_size);
#  129|   
#  130|       /* Set owner bit for all CQEs, so that CQE would look like it is in HW
```

And the same for values from `[32, 64]` range

Reply:
maybe change 1 to 1ul in ib_mlx5.c:128 ?

Reply:
but then it will complain that we compare 64-bit unsigned (`1<<mlx5_cq->cqe_size_log`) and 32-bit unsigned (`cqe_size`) values

Reply:
> maybe change 1 to 1ul in ib_mlx5.c:128 ?

and make `cqe_size` `uint64_t`?

Reply:
i don't think it will complain, it will promote the 32 to 64-bit
it complains when comparing signed and unsigned (because it's not clear how to promote the value without potentially changing it)

Reply:
@yosefe thank you!
I don't see this error with the fix when running csmock